compiler/rustc_parse/src/parser/pat.rs

```diff
@@ -313,6 +313,9 @@ impl<'a> Parser<'a> {
             let pat = self.parse_pat_with_range_pat(false, None)?;
             self.sess.gated_spans.gate(sym::box_patterns, lo.to(self.prev_token.span));
             PatKind::Box(pat)
+        } else if self.eat_keyword(kw::Const) {
+            // Parse `const pat`
+            PatKind::ConstBlock(self.parse_anon_const(lo.to(self.token.span))?)
```

Comment:
shouldn't we also use `self.parse_block()?` here, similar to `parse_anon_const`(which should be called `parse_inline_const` imo)

Reply:
Yeah, I've changed this a while ago but haven't pushed yet.